sycl/test-e2e/Matrix/element_wise_all_ops_gather_scatter.cpp

```diff
@@ -0,0 +1,14 @@
+//==---- element_wise_all_ops_gather_scatter.cpp - DPC++ joint_matrix ------==//
```

Comment:
`element_wise_all_ops_scalar` would be a better name. Gather scatter can imply we are doing gather or scatter type of all to all communications.